upup/pkg/fi/cloudup/awstasks/sshkey.go

```diff
@@ -208,7 +208,7 @@ func (_ *SSHKey) RenderCloudformation(t *cloudformation.CloudformationTarget, a,
 
 	klog.Warningf("Cloudformation does not manage SSH keys; pre-creating SSH key")
 
-	a, err := e.find(cloud)
+	_, err := e.find(cloud)
```

Comment:
this `a` variable is used by line 216, correct? I don't think replacing it with `_` is the correct fix here.

Reply:
this instance shouldn't get flagged – because it is only being set conditionally, in one branch.or can you give me some advice,please?

Reply:
The mechanical fix would be to change the parameter `a` to `_`. But then, why is there a parameter? Is the `e.find(cloud)` call needed?

Reply:
This function is called using reflection, so changing its signature may not be possible:

https://github.com/kubernetes/kops/blob/5cb2d3605c15e28fa077c1b9223e5ef357538086/upup/pkg/fi/context.go#L218-L223

Renaming the parameter to `_` might be the best approach. I'm sure there are other tasks that have unused `a` parameters in some of their Render functions, perhaps we look at how it is handled elsewhere?

Reply:
My question was more along the lines of "Is the value of the `a` parameter semantically different than the `a` obtained from `e.find(cloud)`? If not, the call wouldn't be needed.

Reply:
After reviewing the [commit](https://github.com/kubernetes/kops/commit/7d68f744f6#diff-cbd08d10644cc285349e5b764cd339eeR279) that added these lines, I understand the context a bit more. For reference, `a` is the "actual" state of the task, `e` is the "expected" state of the task.

For CloudFormation (and Terraform) targets we dont care about the actual state of the resource in AWS, we're generating the same terraform output regardless. Therefor most tasks' RenderTerraform functions don't use their `a` parameters. SSHKey is an exception because CloudFormation doesnt support creating them, so Kops needs to create the key AWS directly, which requires knowing the actual state of the ssh key. Now the question becomes "does the `a` parameter get populated with the true actual state of the resource even for cloudformation targets?" and based on some testing I believe the answer is yes. Therefor I think we can get rid of this line and the error checking directly below it.

Reply:
Sorry I misspoke, `a` does not get populated with cloudformation target, therefor the best fix would be to do what [other tasks do](https://github.com/kubernetes/kops/blob/a5ea3809fc76d052f0541a90a0ac2d29294ad984/upup/pkg/fi/cloudup/awstasks/dnszone.go#L317) and use a different variable name, perhaps `keypair`.

Reply:
@rifelpet Hi!I have followed your suggestion and it occurs staticcheck fault.I have already type make and hack/verify-staticcheck.sh and they are passed.Can you tell me if you know about info about it,please?